Feature/sprint14

ElanyFly opened the pull request.

ElanyFly committed:
with binding

ElanyFly committed:
text input delay

ElanyFly committed:
added progress bar to xml

ElanyFly committed:
pr bar show method, fixed visibility issues after clear icon tap

ElanyFly committed:
pause button, track data - previewUrl

ElanyFly committed:
media player added

ElanyFly committed:
смена кнопки пауза\плей в зависимости от состояния.

ElanyFly committed:
Получение данных о времени воспроизведения
Енам класс для состояний плеера.

ElanyFly committed:
Поправлено время при завершении трека

DmitrijEgorow commented:
- [ ] 	Если было activity paused, то кнопка не работает, ничего не воспроизводится

https://github.com/user-attachments/assets/c12ff4a2-991b-44da-b352-639ef78eca12

DmitrijEgorow commented:
- [ ] Приложение аварийно завершается при возврате назад с экрана плеера

https://github.com/user-attachments/assets/7b576804-d801-4207-8ed1-88430723078d
```

FATAL EXCEPTION: main  
Process: com.example.playlistmaker, PID: 7698  
java.lang.IllegalStateException  
at android.media.MediaPlayer.getCurrentPosition(Native Method)  
at com.example.playlistmaker.AudioplayerActivity.getCurrentTrackPosition(AudioplayerActivity.kt:168)  
at com.example.playlistmaker.AudioplayerActivity.timeRunnable$lambda$0(AudioplayerActivity.kt:38)  
at com.example.playlistmaker.AudioplayerActivity.[r8](tex://r8)lambda$0T8HdxUmeiWqF81BYTF7jA-qUNQ(Unknown Source:0)  
at com.example.playlistmaker.AudioplayerActivity$$ExternalSyntheticLambda5.run(Unknown Source:2)  
at android.os.Handler.handleCallback(Handler.java:938)  
at android.os.Handler.dispatchMessage(Handler.java:99)  
at android.os.Looper.loop(Looper.java:223)
```

DmitrijEgorow commented on app/src/main/java/com/example/playlistmaker/AudioplayerActivity.kt:
```diff
@@ -38,37 +54,53 @@ class AudioplayerActivity : AppCompatActivity() {
+            audioGenre.text = track.primaryGenreName
+            audioCountry.text = track.country
+
+//            trackTimeInProgress.text = track.trackTime.toLong().convertMS()
```
Закомментированный код лучше не вливать в master

ElanyFly committed:
Update AudioplayerActivity.kt

исправлен баг с не проигрывающим плеером. после сворачивания приложения до запуска музыки

ElanyFly committed:
добавила хэндлер на клики по трекам
